src/server/app.js

```diff
@@ -28,14 +33,85 @@ if (process.env.NODE_ENV === 'production-notugly') {
   )
 }
 
-const staticHandler = express.static(path.resolve(__dirname, '../../', 'build'))
+const hbs = exphbs.create({
+  // Specify helpers which are only registered on this instance.
+  extname: '.hbs',
+  helpers: {
+    foo: function () {
+      return 'FOO!'
+    },
+    bar: function () {
+      return 'BAR!'
+    }
+  }
+})
 
-app.use(staticHandler)
-app.use('/embed', staticHandler)
-app.get('/embed/:id', videoRoute)
-app.get('/oembed', oembedRoute)
-app.get('*', (req, res) => {
-  res.sendFile(path.resolve(__dirname, '../../', 'build', 'index.html'))
+app.engine('.hbs', hbs.engine)
+app.set('view engine', 'hbs')
+app.set('views', 'src/views')
+app.use(express.static(path.resolve(__dirname, '../../', 'build')))
+
+app.get('/embed/:id', async function (req, res, next) {
+  const { id } = req.params
+  const video = await paratii.core.vids.get(id)
+  let meta = ''
+
+  if (video !== null) {
+    meta = '<link rel="stylesheet" type="text/css" href="/embed/index.css">'
+    meta = routeHelper.openGraphHead(meta, video)
+    meta = routeHelper.twitterCardHead(meta, video)
```

Comment:
Why is `meta` reassigned 3x? Couldn't this be done in one operation somehow?

Reply:
using a partial template maybe?